src/h5web/vis-packs/core/utils.ts

```diff
@@ -365,3 +365,30 @@ export function getSliceSelection(
   // Create slice selection string from dim mapping - e.g. [0, 'y', 'x'] => "0,:,:"
   return dimMapping.map((dim) => (isAxis(dim) ? ':' : dim)).join(',');
 }
+
+export function computeSameRatioDomain(
+  availableSize: Size,
+  abscissaDomain: Domain,
+  ordinateDomain: Domain
+): [Domain, Domain] {
+  const { width, height } = availableSize;
+
+  const abscissaRatio = width / (abscissaDomain[1] - abscissaDomain[0]);
+  const ordinateRatio = height / (ordinateDomain[1] - ordinateDomain[0]);
```

Comment:
The `ratios` here are the number of pixels per coordinate.

Keeping the aspect ratio means that going from 0 to 1 on the abscissa and on the ordinate must span the same number of pixels. This is, both ratios must be equal.

To make the figure fit in the resulting domains, we need to take the **smaller** ratio as common value (i.e. increase the coordinates).